File: src/utils/audio_manager.py
import pygame
import os
from typing import Dict

class AudioManager:
    def __init__(self, config: Dict = None):
        self.sounds: Dict[str, pygame.mixer.Sound] = {}
        self.music_volume = 0.5
        self.sfx_volume = 0.5

        if config:
            self.music_volume = config.get("music_volume", 0.5)
            self.sfx_volume = config.get("sfx_volume", 0.5)

        self.initialized = False
        try:
            pygame.mixer.init()
            self.initialized = True
            print("Audio initialized.")
        except pygame.error as e:
            print(f"Audio initialization failed: {e}")

    def load_sound(self, name: str, filepath: str):
        if not self.initialized or not os.path.exists(filepath):
            return
        try:
            sound = pygame.mixer.Sound(filepath)
            sound.set_volume(self.sfx_volume)
            self.sounds[name] = sound
        except pygame.error as e:
            print(f"Failed to load sound {name}: {e}")

    def play_sound(self, name: str):
        if not self.initialized: return
        if name in self.sounds:
            self.sounds[name].play()
        else:
            # Placeholder logging for development
            # print(f"Playing sound: {name}")
            pass

    def play_music(self, filepath: str, loops: int = -1):
        if not self.initialized or not os.path.exists(filepath):
            # print(f"Playing music: {filepath}")
            return
        try:
            pygame.mixer.music.load(filepath)
            pygame.mixer.music.set_volume(self.music_volume)
            pygame.mixer.music.play(loops)
        except pygame.error as e:
            print(f"Failed to play music {filepath}: {e}")

    def stop_music(self):
        if self.initialized:
            pygame.mixer.music.stop()

File: java_port/core/src/main/java/com/yukkuri/game/engine/AudioManager.java
package com.yukkuri.game.engine;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Music;
import com.badlogic.gdx.audio.Sound;

import java.util.HashMap;
import java.util.Map;

public class AudioManager {
    private final Map<String, Sound> sounds = new HashMap<>();
    private Music currentMusic;
    private float volume = 0.5f;

    public void loadSound(String name, String filepath) {
        try {
            Sound sound = Gdx.audio.newSound(Gdx.files.internal(filepath));
            sounds.put(name, sound);
        } catch (Exception e) {
            Gdx.app.error("AudioManager", "Failed to load sound: " + filepath, e);
        }
    }

    public void playSound(String name) {
        Sound sound = sounds.get(name);
        if (sound != null) {
            sound.play(volume);
        }
    }

    public void setVolume(float volume) {
        this.volume = Math.max(0f, Math.min(1f, volume));
        if (currentMusic != null) {
            currentMusic.setVolume(this.volume);
        }
    }

    public void dispose() {
        for (Sound sound : sounds.values()) {
            sound.dispose();
        }
        if (currentMusic != null) {
            currentMusic.dispose();
        }
    }
}
